optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectEditorViewImpl.java

```diff
@@ -168,8 +177,18 @@ public boolean isFieldPickerEmpty() {
     }
 
     @Override
-    public void initPlanningSolutionScoreTypeOptions( List<Pair<String, String>> options, String selectedScoreType ) {
-        UIUtil.initList( planningSolutionScoreTypeSelector, options, selectedScoreType, false );
+    public void initPlanningSolutionScoreTypeOptions(List<Pair<String, String>> optionPairs) {
+        for (Pair<String, String> optionPair : optionPairs) {
+            Option option = createOption(optionPair);
+            planningSolutionScoreTypeSelector.add(option);
+        }
+    }
+
+    private Option createOption(final Pair<String, String> optionPair) {
+        Option option = (Option) document.createElement("option");
```

Comment:
@manstis The option list is initialized only once during object editor bean initialization. A good point anyway - in my first implementation I hit exactly the same problem you described.